iroh-net/src/dns.rs

```diff
@@ -97,10 +100,69 @@ pub trait ResolverExt {
         host: N,
         timeout: Duration,
     ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
+
+    /// Looks up node info by DNS name.
+    fn lookup_by_domain(&self, name: &str) -> impl Future<Output = Result<NodeAddr>>;
+
+    /// Looks up node info by [`NodeId`] and origin domain name.
+    fn lookup_by_id(
+        &self,
+        node_id: &NodeId,
+        origin: &str,
+    ) -> impl Future<Output = Result<NodeAddr>>;
+
+    /// Perform an ipv4 lookup with a timeout in a staggered fashion.
+    ///
+    /// The first call is done immediately, with added calls according to the staggering strategy.
```

Comment:
Describing `delays_ms` a little more would be helpful.  E.g. are they all  offsets from the time this function is called, or are they offsets from the last request?

Reply:
I know you are right but I'm struggling to convey that these are not offsets in a way that is not terribly verbose. Any suggestions?